src/test/scala/mesosphere/mesos/TaskBuilderTest.scala

```diff
@@ -329,6 +373,9 @@ class TaskBuilderTest extends MarathonSpec with Matchers {
       .build
     TextFormat.shortDebugString(networkInfos.head) should equal(TextFormat.shortDebugString(networkInfoProto))
     networkInfos.head should equal(networkInfoProto)
+
+    taskInfo.hasDiscovery should be (true)
```

Comment:
Yes, that's why I added the following to `assertTaskInfo`, in order to check this in existing tests that use that method:

```scala
    assert(taskInfo.hasDiscovery)
    val discoveryInfo = taskInfo.getDiscovery
    val discoveryInfoProto = MesosProtos.DiscoveryInfo.newBuilder
      .setVisibility(MesosProtos.DiscoveryInfo.Visibility.FRAMEWORK)
      .setName(taskInfo.getName)
      .setPorts(
        MesosProtos.Ports.newBuilder
          .addPorts(MesosProtos.Port.newBuilder.setNumber(8080).setProtocol("tcp"))
          .addPorts(MesosProtos.Port.newBuilder.setNumber(8081).setProtocol("tcp"))
      ).build
    TextFormat.shortDebugString(discoveryInfo) should equal(TextFormat.shortDebugString(discoveryInfoProto))
    discoveryInfo should equal(discoveryInfoProto)
```